pkg/clients/etcd.go

```diff
@@ -0,0 +1,40 @@
+package clients
+
+import (
+	"fmt"
+	"time"
+
+	logging "github.com/op/go-logging"
+
+	"github.com/coreos/etcd/client"
+)
+
+type EtcdConfig struct {
```

Comment:
> Speaking of clients.() methods, do we want any of these to be singletons? 

singleton object I think that we just create them during initialization of application with the configuration of the clients. Allows us to error early if something is going to go wrong imo.

> I think I would have preferred for the PRs to have come in as one unit. Made it difficult to review the set since it looks like C builds on B builds on A, and they all introduce new features on top of the others.
 
wish that at least on comments that told us the order, which could be added.

> We should get rid of [this struct](https://github.com/rthallisey/ansible-service-broker/blob/fe5a7c1ee999858d4d30bd55bfc1761f7357b26f/pkg/apb/client.go#L49)
:+1: 

> I like the clients pkg a lot, decouples things in a way that makes changes a lot less brittle (client init code isn't just thrown about where it's used; it's got a home). Also going to make it real easy and natural to add an Origin() client.
:+1: 

>  Semantically, it's the part of the apb package that's responsible for actually executing apb methods. Could almost be apb/apb.go? Curious what you guys think.

To me, it looks like a lot of the ABP code is in their own files. and client.go should all be in the new package. I am wondering what methods you were thinking should be left in apb?